### PR TITLE
Added functionality to generate PD Contracts

### LINKED ISSUE #1826: As a user I should be able to generate a private data smart contract
The user should be able to use the p.d contract generator to generate a private data smart contract.

- [x] Command Palette command

- [x] Contract is generated 

- [x] Relevant tests are generated 

- [x] Basic PDC is generated 

- [x] Cucumber Tests

### PR DESCRIPTION
**Pull request by au-shah**
Will only work once Lynn's generator changes get merged.

closes #1826 

Signed-off-by: Akshat Shah akshatshah@akshats-mbp.hursley.uk.ibm.com

### CODE REVIEW

File: packages/blockchain-extension/test/commands/UserInputUtil.test.ts [hunk 1]
```diff
@@ -2182,6 +2182,32 @@ describe('UserInputUtil', () => {
         });
     });
 
+    describe('showQuickPickItem', () => {
```
Comment (cazfletch) [inline packages/blockchain-extension/test/commands/UserInputUtil.test.ts]: you don't need these tests, the function already has tests

**Comment by au-shah** [comment]
Made some changes to the existing tutorials. 
Added an extra step `For the scope of this tutorial, choose the `Default Contract` option.` to cater for the private data changes whenever a new project is created.

File: packages/blockchain-extension/tutorials/developer-tutorials/createNewIdentityAttributes.md [hunk 2]
```diff
@@ -48,17 +48,19 @@ As an example, suppose we have a smart contract which allows participants to rec
 
 > Command Palette alternative: `Create New Project`
 
-3. You will first be asked to choose a smart contract language, choose `Typescript`
+3. For the scope of this tutorial, choose the `Default Contract` option.
 
-4. The extension will ask you if you want to name the asset in the generated contract. Next the extension will ask you for an asset name, for this tutorial enter `Car`.
+4. You will first be asked to choose a smart contract language, choose `Typescript`.
```
Comment (cazfletch) [inline packages/blockchain-extension/tutorials/developer-tutorials/createNewIdentityAttributes.md]: choosing the language is now no longer the first step

File: packages/blockchain-extension/tutorials/developer-tutorials/createNewIdentityAttributes.md [hunk 3]
```diff
@@ -48,17 +48,19 @@ As an example, suppose we have a smart contract which allows participants to rec
 
 > Command Palette alternative: `Create New Project`
 
-3. You will first be asked to choose a smart contract language, choose `Typescript`
+3. For the scope of this tutorial, choose the `Default Contract` option.
```
Comment (cazfletch) [inline packages/blockchain-extension/tutorials/developer-tutorials/createNewIdentityAttributes.md]: Change this to "For this tutorial, choose the `Default Contract` option. "

File: packages/blockchain-extension/tutorials/developer-tutorials/debug.md [hunk 4]
```diff
@@ -23,25 +23,27 @@ For the purposes of this tutorial, we'll use TypeScript as the example language.
...
 1. In the left sidebar, click on the __IBM Blockchain Platform__ icon (it looks like a square, and will probably be at the bottom of the set of icons if this was the latest extension you installed!)
 
-2. Mouse-over the `SMART CONTRACT PACKAGES` panel, click the `...` menu, and select `Create Smart Contract Project` from the dropdown.
+2. Mouse-over the `SMART CONTRACT PACKAGES` panel, click the `...` menu, and select `Create New Project` from the dropdown.
 
-> Command Palette alternative: `Create Smart Contract Project`
+> Command Palette alternative: `Create New Project`
 
-3. Choose a smart contract language. JavaScript, TypeScript, Java and Go are all available. For the purpose of this tutorial, please choose `TypeScript`.
+3. For the scope of this tutorial, choose the `Default Contract` option.
```
Comment (cazfletch) [inline packages/blockchain-extension/tutorials/developer-tutorials/debug.md]: change this to "For this tutorial, choose the `Default Contract` option"

File: packages/blockchain-extension/tutorials/ibm-blockchain-platform-vscode-smart-contract/local-dev.md [hunk 5]
```diff
@@ -30,19 +30,21 @@ For the purposes of this tutorial, we'll use TypeScript as the main example lang
 
    > Command Palette alternative: `Create New Project`
 
-3. Choose a smart contract language. JavaScript, TypeScript, Java and Go are all available. This tutorial will be easiest to follow if you choose `TypeScript` or `Java` (please remember to expand the Java sections if you choose Java).
+3. For the scope of this tutorial, choose the `Default Contract` option. The `Private Data Contract` will be covered in a future tutorial.
```
Comment (cazfletch) [inline packages/blockchain-extension/tutorials/ibm-blockchain-platform-vscode-smart-contract/local-dev.md]: change this to "For this tutorial, choose.....

**Pull request closed without merging**